test/systemtests/kubeadm_test.go

```diff
@@ -0,0 +1,981 @@
+		return str, err
+	}
+
+	dd := data.(map[string]interface{})
+	adb := dd["AgentDb"].(map[string]interface{})
+	for key := range adb {
+		actAgents[key] = 1
+	}
+
+	// build expected agentRpc
+	rpcSet := []string{":9002", ":9003"}
+	expAgents := make(map[string]uint32)
+	for agent := range agentIPs {
+		for _, rpc := range rpcSet {
+			k := agent + rpc
+			expAgents[k] = 1
+		}
+	}
+
+	for agent := range expAgents {
+		_, found := actAgents[agent]
+		if !found {
+			return str, errors.New("Agent " + agent + " not found")
+		}
+	}
+
+	// verify there are no extraneous Agents
+	for agent := range actAgents {
+		_, found := expAgents[agent]
+		if !found {
+			return str, errors.New("Unexpected Agent " + agent + " found on " + k.node.Name())
+		}
+	}
+
+	return "", nil
+}
+
+func (k *kubePod) verifyVTEPs(expVTEPS map[string]bool) (string, error) {
+	var data interface{}
+	actVTEPs := make(map[string]uint32)
+	if k.node.Name() == "k8master" {
+		return "", nil
+	}
+	// read vtep information from inspect
+	cmd := "curl -s localhost:9090/inspect/driver | python -mjson.tool"
+	str, err := k.node.tbnode.RunCommandWithOutput(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	err = json.Unmarshal([]byte(str), &data)
+	if err != nil {
+		logrus.Errorf("Unmarshal error: %v", err)
+		return str, err
+	}
+
+	drvInfo := data.(map[string]interface{})
+	vx, found := drvInfo["vxlan"]
+	if !found {
+		logrus.Errorf("vxlan not found in driver info")
+		return str, errors.New("vxlan not found in driver info")
+	}
+
+	vt := vx.(map[string]interface{})
+	v, found := vt["VtepTable"]
+	if !found {
+		logrus.Errorf("VtepTable not found in driver info")
+		return str, errors.New("VtepTable not found in driver info")
+	}
+
+	vteps := v.(map[string]interface{})
+	for key := range vteps {
+		actVTEPs[key] = 1
+	}
+
+	// read local ip
+	localVtep := ""
+	l, found := vt["LocalIp"]
+	if found {
+		switch l.(type) {
+		case string:
+			localVtep = l.(string)
+			actVTEPs[localVtep] = 1
+		}
+	}
+
+	for vtep := range expVTEPS {
+		_, found := actVTEPs[vtep]
+		if !found {
+			return str, errors.New("VTEP " + vtep + " not found")
+		}
+	}
+
+	// verify there are no extraneous VTEPs
+	for vtep := range actVTEPs {
+		_, found := expVTEPS[vtep]
+		if !found {
+			return str, errors.New("Unexpected VTEP " + vtep + " found on " + localVtep)
+		}
+	}
+
+	return "", nil
+}
+func (k *kubePod) verifyEPs(epList []string) (string, error) {
+	// read ep information from inspect
+	if k.node.Name() == "k8master" {
+		return "", nil
+	}
+	cmd := "curl -s localhost:9090/inspect/driver | python -mjson.tool"
+	str, err := k.node.tbnode.RunCommandWithOutput(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	for _, ep := range epList {
+		if !strings.Contains(str, ep) {
+			return str, errors.New(ep + " not found on " + k.node.Name())
+		}
+	}
+
+	return "", nil
+}
+
+//FIXME: This needs to be moved to node abstraction implmentation Once
+//that change is in.
+func (k *kubePod) reloadNode(n *node) error {
```

Comment:
This is the actual vagrant box being reloaded. I have no history on why this was skipped for k8master. My guess is that - for legacy cases, the boxes were prebaked to be master and slaves and reloading master would cause the cluster to be broken. @neelimamukiri: What happens if the kubernetes master node is reloaded? Does any other node take over as master?